spring-observability-core/src/main/java/org/springframework/observability/tracing/listener/TracingInstantRecorder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.observability.tracing.listener;
+
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.observability.event.instant.InstantRecording;
+import org.springframework.observability.tracing.Span;
+import org.springframework.observability.tracing.Tracer;
+
+class TracingInstantRecorder {
+
+	private final Tracer tracer;
+
+	TracingInstantRecorder(Tracer tracer) {
+		this.tracer = tracer;
+	}
+
+	void record(InstantRecording instantRecording) {
+		Span span = this.tracer.currentSpan();
+		if (span != null) {
+			String name = instantRecording.getHighCardinalityName();
+			if (instantRecording.eventNanos() != null) {
+				span.event(TimeUnit.NANOSECONDS.toMicros(instantRecording.eventNanos()), name);
+			}
+			else {
+				span.event(name);
+			}
```

Comment:
Check the latest changes, now the event always has the wall time when it happened so we can do (we don't need the if-else):
```java
span.event(TimeUnit.NANOSECONDS.toMicros(instantRecording.getWallTime()), instantRecording.getHighCardinalityName());
```

```suggestion
			span.event(TimeUnit.NANOSECONDS.toMicros(instantRecording.getWallTime()), instantRecording.getHighCardinalityName());
```